src/components/HomePage/SearchArea/SearchArea.js

```diff
@@ -35,20 +40,98 @@ export const AdvancedSearchController = ({
+    const location = useLocation();
+
+    const history = useHistory();
+
+    const queryParams = useMemo(() => qs.parse(location.search, {
+        ignoreQueryPrefix: true,
+    }), [location]);
+
+    // need to throttle down calling 'history.replace' because
+    // too many invocations might cause the browser to stop responding
+    // value of 350 derived from https://github.com/PoorBillyPilgrim/artcrawl-digital-exhibit/issues/20#issue-761626379
+    // according to this 1000/3 is the lower bound for the 'wait' value
+    const changeURLFilters = useCallback(throttle((location, history, queryParams, changes) => {
+        const newQueryParams = {
+            ...queryParams,
+            ...changes,
+        };
+
+        history.replace({
+            ...location, search: qs.stringify(newQueryParams, {
+                skipNulls: true,
+                arrayFormat: "brackets",
+            }),
+        });
+    }, 350), []);
+
+    const clearURLFilters = useCallback(
+        throttle(
+            (location, history) => history.replace({ ...location, search: "" }),
+            350
+        ), []);
+
+    // destructure input here so mapDispatchToProps.setJobType is 'normalized' with respect to the other dispatch functions
+    const actualSetJobType = useCallback(({ target: { value: jobType } }) => {
+        changeURLFilters(location, history, queryParams, { jobType });
+
+        setJobType(jobType);
+    }, [changeURLFilters, location, history, queryParams, setJobType]);
+
+    const actualSetJobDuration = useCallback((unused, duration) => {
+
+        const [jobMinDuration, jobMaxDuration] = duration;
+
+        changeURLFilters(location, history, queryParams, { jobMinDuration, jobMaxDuration });
+
+        setJobDuration(unused, duration);
+    }, [changeURLFilters, location, history, queryParams, setJobDuration]);
+
+    const actualSetShowJobDurationSlider = useCallback((showJobDurationSlider) => {
+        if (!showJobDurationSlider)
+            changeURLFilters(location, history, queryParams, { jobMinDuration: null, jobMaxDuration: null });
+
+        setShowJobDurationSlider(showJobDurationSlider);
+    }, [changeURLFilters, history, location, queryParams, setShowJobDurationSlider]);
+
+    const actualSetFields = useCallback((fields) => {
+
+        changeURLFilters(location, history, queryParams, { fields });
+
+        setFields(fields);
+    }, [changeURLFilters, history, location, queryParams, setFields]);
+
+    const actualSetTechs = useCallback((technologies) => {
+
+        changeURLFilters(location, history, queryParams, { technologies });
+
+        setTechs(technologies);
+    }, [changeURLFilters, history, location, queryParams, setTechs]);
+
+    const actualResetAdvancedSearchFields = useCallback(() => {
+        clearURLFilters(location, history);
+
+        resetAdvancedSearchFields();
+    }, [clearURLFilters, history, location, resetAdvancedSearchFields]);
+
```

Comment:
This is too verbose to be within the component code, it may be better to extract to a custom hook of ours, and just call the hook here and make it return whatever data we need